docs/rfcs/0039-commonalize-cloud-service-interface.md

```diff
@@ -0,0 +1,316 @@
+| ---------------------------- | ------------- | ------------------------------------------------------------ | --------------------------- |
+| Order guarantee(optional)    | Yes           | Yes                                                          | Yes                         |
+| Topic                        |               | 10,0000 -> 1,000                                             |                             |
+| Subscription                 |               | 12,500,000 -> 100                                            |                             |
+| API throttling(Tokyo region) |               | 1500 -> 300 transactions per second (3000 if high throughput) |                             |
+| messages only once           |               | ◯                                                            |                             |
+
+**Important notice about enabling order guarantee**
+
+* GCP
+  * Possibility of increasing latency
+  * Increase the number of duplicates
+  * We have to design the ordering key so that granularity is not increased.
+* AWS
+  * By default, it limits up to 300 messages per second.
+    * However, it can be configured to handle 6000 messages by enabling throughput mode.
+* Azure
+  * We have to design the session id so that granularity is not increased.
+
+In conclusion, we can enable an order guarantee because we can set the key for each evaluation(userEvaluations + goalEvaluations). In short, we can design the key with fine granularity.
+
+### Cloud SQL & Memorystore
+
+![event-pipeline](./images/0039-image3.png)
+
+#### Comparison Table
+
+|      | GCP        | AWS                                                     | Azure                                                      |
+| ---- | ---------- | ------------------------------------------------------- | ---------------------------------------------------------- |
+|      | Cloud  SQL | Amazon Relational Database Service (RDS), Amazon Aurora | Azure Database for MySQL and Azure Database for PostgreSQL |
+
+|        | GCP         | AWS                | Azure                  |
+| ------ | ----------- | ------------------ | ---------------------- |
+|        | Memorystore | Amazon ElastiCache | Azure Cache            |
+| SLA(%) | >=99.9      | >=99.9             | >=99.9 (from Standard) |
+
+#### Controversial topic
+
+##### 1. Can we configure Memorystore as optional?
+
+Yes. We can configure memory store as a optional in YAML file.
+
+### Druid (GCS) & Kafka
+
+![data-pipeline](./images/0039-image4.png)
+
+We use Druid as a relay DB and usual DB(fething data directory) and Kafka as an intereface for Druid. The amount of data is a huge and we have to handle them as a high performance when fething data directory.
+
+#### Controversial topic
+
+##### 1. Can we stop using Druid & Kafka?
+
+We have the following problems when using Druid & Kafka.
```

Comment:
To support the multi-cloud platform, we will need to decide which DB will replace the Druid first, but we still don't know which DB will fit our usage. I would like to discuss this with you when you have time tomorrow.